### PR TITLE
refactor(Zoe, dapp-svelte-wallet): remove table abstraction from Zoe, dapp-svelte-wallet

### LINKED ISSUE #1494: [Zoe] Review and probably remove table abstraction
This whole function seems rather useless. It is only called in one place, in zoe.js as
```js
      const instance = makeHandle('InstanceHandle');
```
with a string literal argument. This function verifies that the string is a string and then doesn't do anything with it. The instance is a key in a weakStore, not a table, anyway.

My first thought was that `Handle<something>` is to be the type of all `Table` handles, but the only use of `Handle` for typing something is two occurrences of `Handle<'InvitationHandle'>`, which refers to a different definition of `Handle`.

LATE REALIZATION:

It looks like all our other uses of the table abstraction have disappeared, and not it is *only* used for the issuerTable. For another day: If we expect only this one use, we'd probably simplify our lives by collapsing out the table level of abstraction and building issuerTable directly on weakStore. We're paying a lot for generality we're no longer using. Our attempt to type both the table level and the issuerTable level probably makes this generality even more of a differential expense.

_Originally posted by @erights in https://github.com/_render_node/MDExOlB1bGxSZXF1ZXN0NDY1NjU3MTI5/timeline/more_items_

### PR DESCRIPTION
**Pull request by katelynsills**
It turns out that the current issuerTable already has one turn boundary, when we wait on `Promise.resolve(issuerP)`. If one turn boundary before saving is acceptable, then we can do away with the entire infrastructure of managing issuer records (no `issuersInProgress`) and perform all of our waiting in one `await Promise.all()`

To answer #1494, `makeHandle` is purely to make the typing work. No other reason.

`dapp-svelte-wallet` has a lot of duplicate code with the issuerTable. One difference is that it also stores a `boardId` for the issuer. I tried to add the boardId storage without having to duplicate the issuerTable code. @michaelfig can you take a look at this part? What is the best way to test dapp-svelte-wallet? There are no tests. 

The `test/swingsetTests/zoe-metering/test-zoe-metering.js` tests do not pass with this change, but I cannot figure out why. An error is thrown: `issuer was using a brand which was not its own`. It is the ZoeIssuer, so this is not the case. Is this an infrastructure bug dealing with presence identity that I should file, or am I missing something?

Closes #1494.

### CODE REVIEW

**Comment by michaelfig** [comment]
> What is the best way to test dapp-svelte-wallet? There are no tests.

Your old wallet and dehydrator tests have been updated.  Run them with `cd packages/dapp-svelte-wallet/api && yarn test`

**Comment by katelynsills** [comment]
> > What is the best way to test dapp-svelte-wallet? There are no tests.
> 
> Your old wallet and dehydrator tests have been updated. Run them with `cd packages/dapp-svelte-wallet/api && yarn test`

Ok, thanks!

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js [hunk 1]
```diff
@@ -366,10 +293,16 @@ export async function makeWallet({
...
-      [...brandMapping.petnameToVal.entries()].map(([petname, brand]) => [
-        petname,
-        brandTable.get(brand),
-      ]),
+      [...brandMapping.petnameToVal.entries()].map(([petname, brand]) => {
+        const issuerRecord = brandTable.getByBrand(brand);
+        return [
+          petname,
+          {
+            issuerBoardId: issuerToBoardId.get(issuerRecord.issuer),
```
Comment (michaelfig) [inline packages/dapp-svelte-wallet/api/src/lib-wallet.js]: I would prefer moving this line below the `...issuerRecord`, just to emphasise we're overriding it.

**Review by michaelfig — COMMENTED** [review]
The wallet components LGTM.  Thanks for the update!

File: packages/zoe/src/zoeService/zoe.js [hunk 2]
```diff
@@ -40,11 +39,25 @@ function makeZoe(vatAdminSvc, zcfBundleName = undefined) {
...
+   * @template {string} H
+   * @param {H} handleType the string literal type of the handle
+   * @returns {Handle<H>}
+   */
+  const makeHandle = handleType => {
+    // This assert ensures that handleType is referenced.
+    assert.typeof(handleType, 'string', 'handleType must be a string');
+    // Return the intersection type (really just an empty object).
+    return /** @type {Handle<H>} */ (harden({}));
+  };
```
Comment (erights) [inline packages/zoe/src/zoeService/zoe.js:59]: Does this function do anything useful?

File: packages/zoe/src/issuerTable.js [hunk 3]
```diff
@@ -4,122 +4,74 @@ import { assert } from '@agoric/assert';
...
+      const brandP = E(issuerP).getBrand();
+      const brandIssuerMatchP = E(brandP).isMyIssuer(issuerP);
+      const amountMathKindP = E(issuerP).getAmountMathKind();
+      /** @type {[Issuer,Brand,boolean,AmountMathKind]} */
+      const [
+        issuer,
+        brand,
+        brandIssuerMatch,
+        amountMathKind,
+      ] = await Promise.all([
```
[the remark this answers is not in the document]
Comment (erights) [reply packages/zoe/src/issuerTable.js:48]: > It turns out that the current issuerTable already has one turn boundary, when we wait on Promise.resolve(issuerP). If one turn boundary before saving is acceptable, 

I don't understand how that's relevant. The above analysis seems independent of the number of turns of delay.

File: packages/zoe/src/zoeService/zoe.js (hunk 2 above)
Comment (erights) [reply packages/zoe/src/zoeService/zoe.js:59]: > To answer #1494, makeHandle is purely to make the typing work. No other reason.

Sorry I missed this. Resolving. (But not happy that typing forces us to add such a silly piece of code.) What is the typing problem this solves?

Comment (michaelfig) [reply packages/zoe/src/zoeService/zoe.js:59]: > What is the typing problem this solves?

It was used to make an `InstanceHandle` and an `InstallationHandle` and any others disjoint types, so we caught errors when one was used in the wrong context.

Comment (erights) [reply packages/zoe/src/zoeService/zoe.js:59]: Do we still have such handles? Why can't we just use object types?

**Comment by katelynsills** [comment]
> Do we still have such handles? Why can't we just use object types?

I think we may have some more internal handles in the near future so there
may be need to distinguish them in the types.

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js [hunk 4]
```diff
@@ -60,80 +58,9 @@ export async function makeWallet({
   /** @type {Mapping<Installation>} */
   const installationMapping = makeMapping('installation');
 
-  /**
-   * Brand Table
-   */
-  const makeBrandTable = () => {
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/dapp-svelte-wallet/api/src/lib-wallet.js:66]: It was just borrowing with the addition of adding the issuerBoardId

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js [hunk 5]
```diff
@@ -366,10 +293,16 @@ export async function makeWallet({
...
-        brandTable.get(brand),
-      ]),
+      [...brandMapping.petnameToVal.entries()].map(([petname, brand]) => {
+        const issuerRecord = brandTable.getByBrand(brand);
+        return [
+          petname,
+          {
+            issuerBoardId: issuerToBoardId.get(issuerRecord.issuer),
+            ...issuerRecord,
+          },
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/dapp-svelte-wallet/api/src/lib-wallet.js]: Sorry, I don't know what you mean here. There's no `get` on the issuerTable/brandTable by design

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js [hunk 6]
```diff
@@ -525,9 +458,9 @@ export async function makeWallet({
   };
 
   const publishIssuer = async brand => {
-    const brandRecord = brandTable.get(brand);
+    const brandRecord = brandTable.getByBrand(brand);
     const issuerBoardId = await E(board).getId(brandRecord.issuer);
-    brandTable.update(brand, { issuerBoardId });
+    issuerToBoardId.set(brandRecord.issuer, issuerBoardId);
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/dapp-svelte-wallet/api/src/lib-wallet.js]: Oh good point

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js (hunk 1 above)
Comment (katelynsills) [reply packages/dapp-svelte-wallet/api/src/lib-wallet.js]: Yup, sounds good

File: packages/zoe/src/issuerTable.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/zoe/src/issuerTable.js:48]: Yes, the above analysis is independent, but that doesn't imply my point is irrelevant. :) I could see someone being concerned (wrongly) that we are adding more asynchrony, so my point is that we have the same number of turn boundaries as before.

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (erights) [reply packages/dapp-svelte-wallet/api/src/lib-wallet.js]: This PR provides `issuerTable.getByBrand`. `brandTable` is an issuerTable. Why can't you use `brandTable.getByBrand(brand)` here where the old code said `brandTable.get(brand)` ?

**Comment by katelynsills** [comment]
I figured out what the "infrastructure" problem was - it's that `E(brand).isMyIssuer()` doesn't take promises, for no good reason. I will make a separate issue and PR for that.

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/dapp-svelte-wallet/api/src/lib-wallet.js]: Oh, I do get the record by doing `const issuerRecord = brandTable.getByBrand(brand);` a few lines above. I think I still don't see the problem

File: packages/zoe/src/issuerTable.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (erights) [reply packages/zoe/src/issuerTable.js:48]: Gotcha. I thought you were making an entirely different point.

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (erights) [reply packages/dapp-svelte-wallet/api/src/lib-wallet.js]: The addition of the `issuerBoardId` caused me to misunderstand the before vs after difference. The before code just has the issuerRecord. Resolving.

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (erights) [reply packages/dapp-svelte-wallet/api/src/lib-wallet.js:66]: Comment cleared up the other confusion.

**Review by erights — APPROVED** [review]
LGTM

**Pull request merged**